src/providers/memory/qgsmemoryprovider.cpp

```diff
@@ -60,6 +60,18 @@ QgsMemoryProvider::QgsMemoryProvider( const QString& uri )
     mWkbType = QgsWkbTypes::MultiLineString;
   else if ( geometry == "multipolygon" )
     mWkbType = QgsWkbTypes::MultiPolygon;
+  else if ( geometry == "point25d" )
```

Comment:
this block should use QgsWkbTypes::parseType() - then we'd also get correct handling of Z/M/curved (+less duplicate code)